chef/automate
# Dan/fix protoc deprecations

### :nut_and_bolt: Description: What code changed, and why?

After updating the golang proto compiler plugin in #4147, compiling the protos in the studio would emit a LOT of these warnings:

```
WARNING: Package "github.com/golang/protobuf/protoc-gen-go/generator" is deprecated.
	A future release of golang/protobuf will delete this package,
	which has long been excluded from the compatibility promise.
```

You could see 2 or 3 of these per compiled file. This patch updates the protobuf plugins to versions that don't trigger these warnings.

### :chains: Related Resources

* The deprecation warning: https://github.com/golang/protobuf/blob/d04d7b157bb510b1e0c10132224b616ac0e26b17/protoc-gen-go/generator/generator.go#L42-L47
* #4147

### :+1: Definition of Done

No deprecation warnings

### :athletic_shoe: How to Build and Test the Change

* Enter the studio. `rm -r /src/bin/*` to clear out existing compiled versions of the protobuf plugins
* `compile_all_protobuf_components`
* Check the output for the deprecation warning
